spring-cloud-function-context/src/main/java/org/springframework/cloud/function/context/catalog/SimpleFunctionRegistry.java

```diff
@@ -819,16 +819,20 @@ private Object fluxifyInputIfNecessary(Object input) {
 
 			if (!this.isRoutingFunction() && !(input instanceof Publisher)) {
 				Object payload = input;
-				if (input instanceof Message) {
-					if (((Message) input).getHeaders().containsKey("payload")) {
-						payload = ((Message) input).getHeaders().get("payload");
+				var treatPayloadAsPlainText = false;
```

Comment:
**[NOTE]** I am not sure if this is the best place to fix the issue or if this will introduce other issues. Submitting the PR to start the discussion/review process though.